gems/smithy/lib/smithy/plan.rb

```diff
@@ -31,5 +33,19 @@ def initialize(model, type, options = {})
 
     # @return [Array<Polish>] The polishes that apply to this plan.
     attr_reader :polishes
+
+    # @return [Hash[String, BuiltInBinding]] Array of all registered builtins
+    attr_reader :built_in_bindings
+
+    # @return [Hash[String, FunctionBinding]] Array of all registered functions
+    attr_reader :function_bindings
```

Comment:
These are not clear that they are endpoint related. I also don't think these should be on plan. Could a view not call the plans welds (that's why weld is an instance variable on plan?) and initialize all of the bindings for the view to use?

Reply:
Yeah - I guess we could push that to views to do when needed.  The issue I ran into is that each instance of EndpointParameter needs the built in bindings, which means I either need to call the weld many times (once per parameter x operations) or build it in the view(s) and pass it around.  It seemed like something that should only get done once and stored.

Reply:
I think it's the view's responsibility to get what it needs (bindings) from welds or whatever sources, and pass it to any sub views if it has to.